src/Query/Condition.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\entity\Query;
+
+/**
+ * Represents a single access condition.
+ */
+final class Condition {
+
+  /**
+   * The supported operators.
+   *
+   * @var array
+   */
+  protected static $supportedOperators = [
+    '=', '<>', '<', '<=', '>', '>=', 'BETWEEN', 'NOT BETWEEN',
+    'IN', 'NOT IN', 'IS NULL', 'IS NOT NULL',
+  ];
+
+  /**
+   * The field.
+   *
+   * @var string
+   */
+  protected $field;
+
+  /**
+   * The value.
+   *
+   * @var mixed
+   */
+  protected $value;
+
+  /**
+   * The operator.
+   *
+   * @var string
+   */
+  protected $operator;
+
+  /**
+   * Constructs a new Condition object.
+   *
+   * @param string $field
+   *   The field, with an optional column name. E.g: 'uid', 'address.locality'.
+   * @param mixed $value
+   *   The value.
+   * @param string $operator
+   *   The operator.
+   *   Possible values: =, <>, <, <=, >, >=, BETWEEN, NOT BETWEEN,
+   *                   IN, NOT IN, IS NULL, IS NOT NULL.
+   */
+  public function __construct($field, $value, $operator = NULL) {
+    // Provide a default based on the data type of the value.
+    if (!isset($operator)) {
+      $operator = is_array($value) ? 'IN' : '=';
```

Comment:
👌

Reply:
I'm wondering whether this is actually the right thing to do. After drupalgeddon Nr. 1 there was this drupal core issue: https://www.drupal.org/node/2358079 which would probably argue this automatic conversion should not be done.